internal/cmd/stack/local_preview.go

```diff
@@ -125,3 +133,51 @@ func localPreview() cli.ActionFunc {
 		return terminal.Error()
 	}
 }
+
+// EnvironmentVariable represents a key-value pair of environment variables
+type EnvironmentVariable struct {
+	Key   graphql.String `json:"key"`
+	Value graphql.String `json:"value"`
+}
+
+func parseEnvVariablesForLocalPreview(cliCtx *cli.Context) ([]EnvironmentVariable, error) {
+	envVars := make([]EnvironmentVariable, 0)
+
+	if !cliCtx.IsSet(flagEnvVars.Name) && !cliCtx.IsSet(flagEnvVarsTF.Name) {
```

Comment:
Don't think this early return is necessary, no? It'll just iterate over two empty lists.